controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -533,3 +551,432 @@ func (r *KafkaUserReconciler) getKafkaUserCredsFromSecret(
+}
+
+func (r *KafkaUserReconciler) handleDeleteCertificate(ctx context.Context, user *v1beta1.KafkaUser, l logr.Logger, certCreationData *v1beta1.CertificateCreationData) error {
+	certID, err := r.getKafkaUserCertIDFromSecret(ctx, certCreationData)
+	if err != nil {
+		l.Error(
+			err, "Cannot get Kafka user certificate ID from secret",
+			"kafka user certificate secret name", certCreationData.SecretName,
+			"kafka user certificate secret namespace", certCreationData.SecretNamespace,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user certificate ID from secret is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	err = r.API.DeleteKafkaUserCertificate(instaclustr.KafkauserCertificatesEndpoint, certID)
+	if err != nil {
+		l.Error(err, "Cannot Delete Kafka user Mtls certificate",
+			"user", user.Name,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.DeletionFailed,
+			"Certificate deletion is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	secret := &v1.Secret{}
+	certSecretNamespacedName := types.NamespacedName{
+		Name:      certCreationData.SecretName,
+		Namespace: certCreationData.SecretNamespace,
+	}
+	err = r.Client.Get(ctx, certSecretNamespacedName, secret)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			l.Error(err, "Cannot get Kafka user certificate secret. Secret is not found",
+				"kafka user certificate secret name", certCreationData.SecretName,
+				"kafka user certificate secret namespace", certCreationData.SecretNamespace,
+			)
+			r.EventRecorder.Eventf(
+				user, models.Warning, models.FetchFailed,
+				"Fetch user certificate secret is failed. Reason: %v",
+				err,
+			)
+
+			return err
+		}
+		l.Error(err, "Cannot get Kafka user certificate secret.",
+			"kafka user certificate secret name", certCreationData.SecretName,
+			"kafka user certificate secret namespace", certCreationData.SecretNamespace,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user certificate secret is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	controllerutil.RemoveFinalizer(secret, models.DeletionFinalizer)
+	err = r.Update(ctx, secret)
+	if err != nil {
+		l.Error(err, "Cannot remove finalizer from secret", "secret name", secret.Name)
+
+		r.EventRecorder.Eventf(user, models.Warning, models.PatchFailed,
+			"Resource patch is failed. Reason: %v", err)
+
+		return err
+	}
+
+	err = r.Client.Delete(ctx, secret)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		l.Error(err, "Cannot delete Kafka user certificate secret",
+			"kafka user certificate secret name", certCreationData.SecretName,
+			"kafka user certificate secret namespace", certCreationData.SecretNamespace,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.DeletionFailed,
+			"Delete user certificate secret is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	l.Info("Kafka user Mtls certificate has been deleted",
```

Comment:
Mtls -> mTLS, refactor in all similar cases, please

Reply:
done